File: python/leetcode/maximum-subarray.py
class Solution:
    def maxSubArray(self, nums):
        start_idx = 0
        max_s = nums[start_idx]
        while(start_idx < len(nums)):
            this_sum = nums[start_idx]
            if (this_sum > max_s):
                max_s = this_sum
            for i in range(start_idx + 1, len(nums)):
                this_sum += nums[i]
                if this_sum > max_s:
                    max_s = this_sum
                elif this_sum < 0:
                    break
            start_idx += 1
        return max_s

File: java/leetcode/maximum-subarray.java
class Solution {
    static int maxSubArray(int[] nums) {
    	int startIdx = 0;
    	int maxS = nums[startIdx];
    	int thisSum = maxS;
    	int i = 0;
    	while (startIdx < nums.length) {
    		thisSum = nums[startIdx];
    		if (thisSum > maxS) {
    			maxS = thisSum;
    		}
    		for (i = startIdx + 1; i < nums.length; i++) {
    			thisSum += nums[i];
    			if (thisSum > maxS) {
    				maxS = thisSum;
    			} else if (thisSum < 0) {
    				break;
    			}
    		}
    		startIdx++;
    	}
    	return maxS;
    }
}
